cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/config/WebSecurity.java

```diff
@@ -10,31 +10,62 @@
 
 package org.zowe.apiml.cloudgatewayservice.config;
 
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Primary;
-import org.springframework.security.config.Customizer;
+import org.springframework.security.authentication.ReactiveAuthenticationManager;
 import org.springframework.security.config.web.server.ServerHttpSecurity;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.core.userdetails.ReactiveUserDetailsService;
 import org.springframework.security.core.userdetails.User;
+import org.springframework.security.web.authentication.preauth.x509.SubjectDnX509PrincipalExtractor;
 import org.springframework.security.web.server.SecurityWebFilterChain;
+import org.zowe.apiml.product.constants.CoreService;
 import reactor.core.publisher.Mono;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @Configuration
 public class WebSecurity {
 
+    @Value("${apiml.security.x509.registry.allowedUsers:}")
+    private List<String> usersAllowList = new ArrayList<>();
+
     @Bean
     public SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity http) {
-        http.x509(Customizer.withDefaults()).csrf().disable()
-            .authorizeExchange(exchange -> exchange.anyExchange().permitAll());
+
+        SubjectDnX509PrincipalExtractor principalExtractor = new SubjectDnX509PrincipalExtractor();
+
+        ReactiveAuthenticationManager authenticationManager = authentication -> {
+            authentication.setAuthenticated(usersAllowList.contains(authentication.getName()));
+            return Mono.just(authentication);
+        };
+
+        http.x509(x509 ->
+                x509
+                    .principalExtractor(principalExtractor)
+                    .authenticationManager(authenticationManager)).authorizeExchange()
+            .pathMatchers("/" + CoreService.CLOUD_GATEWAY.getServiceId() + "/api/v1/registry/**").authenticated()
+            .and().csrf().disable()
+            .authorizeExchange().anyExchange().permitAll();
+
         return http.build();
     }
 
     @Bean
     @Primary
     ReactiveUserDetailsService userDetailsService() {
-        return username -> Mono.just(User.withUsername(username).password("password").authorities("user").build());
-    }
 
+        return username -> {
+            List<GrantedAuthority> authorities = new ArrayList<>();
+            if (usersAllowList.stream().anyMatch(allowedUser -> allowedUser != null && allowedUser.equalsIgnoreCase(username))) {
```

Comment:
This implementation is different than I suggested. The suggested one allowed to use `*` as the config value and then all users (with a certificate in the trust store) have the role...

Reply:
Sure, I've added it now. Your dream have come true!